app/src/main/java/com/example/riden/activities/directionhelpers/FetchURL.java

```
@@ -0,0 +1,79 @@
+    Context mContext;
+    String directionMode = "driving";
+
+    public FetchURL(Context mContext) {
+        this.mContext = mContext;
+    }
+
+    @Override
+    protected String doInBackground(String... strings) {
+        // For storing data from web service
+        String data = "";
+        directionMode = strings[1];
+        try {
+            // Fetching the data from web service
+            data = downloadUrl(strings[0]);
+            Log.d("mylog", "Background task data " + data.toString());
+        } catch (Exception e) {
+            Log.d("Background Task", e.toString());
+        }
+        return data;
+    }
+
+    @Override
+    protected void onPostExecute(String s) {
+        super.onPostExecute(s);
+        PointsParser parserTask = new PointsParser(mContext, directionMode);
+        // Invokes the thread for parsing the JSON data
+        parserTask.execute(s);
+    }
+
+    private String downloadUrl(String strUrl) throws IOException {
+        String data = "";
+        InputStream iStream = null;
+        HttpURLConnection urlConnection = null;
+        try {
+            URL url = new URL(strUrl);
+            // Creating an http connection to communicate with url
+            urlConnection = (HttpURLConnection) url.openConnection();
+            // Connecting to url
+            urlConnection.connect();
+            // Reading data from url
+            iStream = urlConnection.getInputStream();
+            BufferedReader br = new BufferedReader(new InputStreamReader(iStream));
+            StringBuffer sb = new StringBuffer();
+            String line = "";
+            while ((line = br.readLine()) != null) {
+                sb.append(line);
+            }
+            data = sb.toString();
+            Log.d("mylog", "Downloaded URL: " + data.toString());
+            br.close();
+        } catch (Exception e) {
+            Log.d("mylog", "Exception downloading URL: " + e.toString());
```

Do you want to silently fail in case of an exception ?